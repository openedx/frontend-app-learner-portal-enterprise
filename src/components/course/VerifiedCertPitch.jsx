import React from 'react';
import PropTypes from 'prop-types';
import { Icon } from '@openedx/paragon';
import { ContentPasteGo, RocketLaunch } from '@openedx/paragon/icons';

import './styles/VerifiedCertPitch.scss';

const VerifiedCertPitchIcon = ({ icon }) => (
  <div className="icon d-flex bg-info mr-3 p-2 align-items-center justify-content-center">
    <Icon className="text-white" src={icon} />
  </div>
);

VerifiedCertPitchIcon.propTypes = {
  icon: PropTypes.shape().isRequired,
};

const VerifiedCertPitch = () => (
  <div className="verified-cert-pitch mb-5">
    <h3 className="mb-4">
      Pursue a Verified Certificate to highlight the knowledge and skills you gain
    </h3>
    <ul className="pl-0 mb-0 list-unstyled">
      <li className="d-flex mb-4">
        <VerifiedCertPitchIcon icon={RocketLaunch} />
        <div>
          <h4>Official and Verified</h4>
          <span>
            Receive an instructor-signed certificate with the institution&apos;s
            logo to verify your achievement.
          </span>
        </div>
      </li>
      <li className="d-flex mb-4">
        <VerifiedCertPitchIcon icon={ContentPasteGo} />
        <div>
          <h4>Easily Shareable</h4>
          <span>
            Add the certificate to your CV or resume, or post it
            directly on LinkedIn.
          </span>
        </div>
      </li>
    </ul>
  </div>
);

export default VerifiedCertPitch;
