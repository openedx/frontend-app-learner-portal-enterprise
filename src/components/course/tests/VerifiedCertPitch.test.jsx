import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';

import VerifiedCertPitch from '../VerifiedCertPitch';

describe('VerifiedCertPitch', () => {
  it('renders', () => {
    const tree = renderer
      .create(<VerifiedCertPitch />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
