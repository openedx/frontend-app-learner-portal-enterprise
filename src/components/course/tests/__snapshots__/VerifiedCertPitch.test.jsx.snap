// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerifiedCertPitch renders 1`] = `
<div
  className="verified-cert-pitch mb-5"
>
  <h3
    className="mb-4"
  >
    Pursue a Verified Certificate to highlight the knowledge and skills you gain
  </h3>
  <ul
    className="pl-0 mb-0 list-unstyled"
  >
    <li
      className="d-flex mb-4"
    >
      <div
        className="icon d-flex bg-info mr-3 p-2 align-items-center justify-content-center"
      >
        <span
          className="pgn__icon text-white"
        >
          <svg
            aria-hidden={true}
            fill="none"
            focusable={false}
            height={24}
            role="img"
            viewBox="0 0 24 24"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.19 6.35c-2.04 2.29-3.44 5.58-3.57 5.89L2 10.69l4.81-4.81 2.38.47zM11.17 17s3.74-1.55 5.89-3.7c5.4-5.4 4.5-9.62 4.21-10.57-.95-.3-5.17-1.19-10.57 4.21C8.55 9.09 7 12.83 7 12.83L11.17 17zm6.48-2.19c-2.29 2.04-5.58 3.44-5.89 3.57L13.31 22l4.81-4.81-.47-2.38zM9 18c0 .83-.34 1.58-.88 2.12C6.94 21.3 2 22 2 22s.7-4.94 1.88-6.12A2.996 2.996 0 0 1 9 18zm4-9c0-1.1.9-2 2-2s2 .9 2 2-.9 2-2 2-2-.9-2-2z"
              fill="currentColor"
            />
          </svg>
        </span>
      </div>
      <div>
        <h4>
          Official and Verified
        </h4>
        <span>
          Receive an instructor-signed certificate with the institution's logo to verify your achievement.
        </span>
      </div>
    </li>
    <li
      className="d-flex mb-4"
    >
      <div
        className="icon d-flex bg-info mr-3 p-2 align-items-center justify-content-center"
      >
        <span
          className="pgn__icon text-white"
        >
          <svg
            aria-hidden={true}
            fill="none"
            focusable={false}
            height={24}
            role="img"
            viewBox="0 0 24 24"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5 5h2v3h10V5h2v6h2V3h-6.18C14.4 1.84 13.3 1 12 1s-2.4.84-2.82 2H3v18h7v-2H5V5zm7-2c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1z"
              fill="currentColor"
            />
            <path
              d="m18.01 13-1.42 1.41 1.58 1.58H12v2h6.17l-1.58 1.59 1.42 1.41 3.99-4z"
              fill="currentColor"
            />
          </svg>
        </span>
      </div>
      <div>
        <h4>
          Easily Shareable
        </h4>
        <span>
          Add the certificate to your CV or resume, or post it directly on LinkedIn.
        </span>
      </div>
    </li>
  </ul>
</div>
`;
